### ISSUE TITLE
[BUG] Darwin error at service stop in Vulture OS

### ISSUE THREAD

**Issue by HugoSoszynski** [VultureProject/darwin#74]
**Describe the bug**
When doing a `service darwin stop`, `CRITICAL` logs pop telling that some filters (not all) cannot open UNIX socket because of a bad weak pointer.

**Plateform (please complete the following information):**
 - OS (version): FreeBSD 12.0
 - Darwin version: 1.0
 - Vulture BASE version: 0.9.7
 - Vulture GUI version: 0.9.92

**To Reproduce**
1. Install Vulture 4
2. Create a new Darwin Policy containing:
  - fconnection
  - finspection
  - fdga
  - fhostlookup
  - ftanomaly
3. Open logs in one terminal (`tail -f /var/log/darwin/darwin.log`)
4. `service darwin stop`

**Expected behavior**
Darwin filters terminating without errors.

**Screenshots/logs**
**Try 1:**
```
{"date":"Wed Oct  2 12:48:49 2019","level":"CRITICAL","filter":"logs_1","message":"Core::run:: Cannot open unix socket: bad_weak_ptr"} 
{"date":"Wed Oct  2 12:48:49 2019","level":"CRITICAL","filter":"content_inspection_2","message":"Core::run:: Cannot open unix socket: bad_weak_ptr"} 
{"date":"Wed Oct  2 12:48:49 2019","level":"CRITICAL","filter":"tanomaly_2","message":"Core::run:: Cannot open unix socket: bad_weak_ptr"} 
```

**Try 2:**
```
{"date":"Wed Oct  2 12:49:34 2019","level":"CRITICAL","filter":"logs_1","message":"Core::run:: Cannot open unix socket: bad_weak_ptr"}
{"date":"Wed Oct  2 12:49:34 2019","level":"CRITICAL","filter":"content_inspection_2","message":"Core::run:: Cannot open unix socket: bad_weak_ptr"}
{"date":"Wed Oct  2 12:49:34 2019","level":"CRITICAL","filter":"dga_2","message":"Core::run:: Cannot open unix socket: bad_weak_ptr"}
{"date":"Wed Oct  2 12:49:34 2019","level":"CRITICAL","filter":"tanomaly_2","message":"Core::run:: Cannot open unix socket: bad_weak_ptr"}
```

**Additional context**
Using Rsyslog's mmdarwin module with an impcap listener.

**Comment by NS4nti** [comment]
I don't succeed in reproducing this bug, can you maybe post your config file ?

**Comment by frikilax** [comment]
additional context, while working on something else.
with logs filter and a new version of redis manager.
filter was launched in DEVELOPER mode, logs are what followed a Ctrl+C.
```
"date":"Thu Oct 17 23:19:28 2019","level":"DEBUG","filter":"logs_1","message":"Server::Handle:: Closing acceptor"}
{"date":"Thu Oct 17 23:19:28 2019","level":"DEBUG","filter":"logs_1","message":"Session::Stop::"}
{"date":"Thu Oct 17 23:19:28 2019","level":"DEBUG","filter":"logs_1","message":"Server::Handle:: Closing acceptor"}
{"date":"Thu Oct 17 23:19:28 2019","level":"INFO","filter":"logs_1","message":"Server::HandleAccept:: Acceptor closed, closing server..."}                                                                                                  
{"date":"Thu Oct 17 23:19:28 2019","level":"DEBUG","filter":"logs_1","message":"Session::ReadHeaderCallback:: Reading header"}                                                                                                              
{"date":"Thu Oct 17 23:19:28 2019","level":"INFO","filter":"logs_1","message":"Monitor::HandleAccept:: Acceptor closed, closing monitor..."}                                                                                                
{"date":"Thu Oct 17 23:19:28 2019","level":"WARNING","filter":"logs_1","message":"Session::ReadHeaderCallback:: Operation canceled"}                                                                                                        
{"date":"Thu Oct 17 23:19:28 2019","level":"CRITICAL","filter":"logs_1","message":"Core::run:: Cannot open unix socket: bad_weak_ptr"}                                                                                                      
{"date":"Thu Oct 17 23:19:28 2019","level":"DEBUG","filter":"logs_1","message":"Core::ClearPID:: Removing PID file"}
```

**Comment by HugoSoszynski** [comment]
It looks like this happen when we close a filter running some callback.

I modified the way a filter is cleaning its resources.
This seems to solve a lot of memory issues when stopping a filter that is running some callbacks.

Seems to solve this issue, needs further testing.
Change will come in v1.1 with addition of the threading _(moving this issue to milestone Version 1.1)_.

**Comment by HugoSoszynski** [comment]
Seems to be resolved in #98.

Was due to early clean of the Sessions.
